### ISSUE TITLE
Missing quoting in generated -pkg.el files breaks Emacs 24.1 and 24.2

### ISSUE THREAD

**Issue by purcell** [melpa/package-build#46]
We currently seem to get multi-file packages that have manifests like this:

```el
(define-package "package-lint" "20200906.512" "A linting library for elisp package authors"
  '((cl-lib "0.5")
    (emacs "24.1")
    (let-alist "1.0.6"))
  :commit "05596996286089acc7693b700c7c31780439e39f" :authors
  (("Steve Purcell" . "steve@sanityinc.com")
   ("Fanael Linithien" . "fanael4@gmail.com"))
  :maintainer
  ("Steve Purcell" . "steve@sanityinc.com")
  :keywords
  ("lisp")
  :url "https://github.com/purcell/package-lint")
;; Local Variables:
;; no-byte-compile: t
;; End:
```

In Emacs 24.1 and 24.2, this seems to produce errors at install time along the lines of
```
Invalid function: ("Steve Purcell" . "steve@sanityinc.com")
```
An example is [here](https://github.com/purcell/page-break-lines/runs/1646639021?check_suite_focus=true#step:4:84).

Should we be quoting those lists? Feels like it.

**Comment by purcell** [comment]
Would `package-desc-from-define` be the place to fix this, @tarsius?

**Comment by tarsius** [comment]
Fixed in dfc69460ea6e13e7c37d1578f75ff8e59727c9fb.

**Comment by purcell** [comment]
Nice, thanks for tracking it down.
